Refs #17 Push通知用のモジュールを追加

ist-ikeno opened the pull request.

ist-ikeno committed:
Refs #17 Push通知用のモジュールを追加

ist-ikeno committed:
Merge branch 'develop' into feature/#17_PushNotification_Module

# Conflicts:
#	Project-iOS/XLProjectName/XLProjectName/AppDelegate+PushNotification.swift

ist-ikeno commented:
@satoshin-IST push通知周りのモジュールを実装いたしましたので、レビューお願いいたします。

SatoshiN303 commented:
@ist-ikeno 了解！

SatoshiN303 commented on Project-iOS/XLProjectName/XLProjectName/Helpers/PushNotification/NotificationPayload.swift:
```diff
@@ -0,0 +1,44 @@
+//  XLProjectName
+//
+//  Created by 池野 湧太 on 2018/02/05.
+//  Copyright © 2018年 'XLOrganizationName'. All rights reserved.
```
Skelton.swift で生成したときに実装者の名前で書き換えたいので以下に変更お願いできますでしょうか。
`Created by 池野 湧太 on 2018/02/05.`  → `Created by XLAuthorName `

他の箇所も同様にお願い致します。

SatoshiN303 commented on Project-iOS/XLProjectName/XLProjectName/Helpers/PushNotification/PushNotificationModule.swift:
```diff
@@ -0,0 +1,224 @@
+        static let snsPlatformArn = ""
+        #endif
+    }
+    
```
環境変数の設定の仕方は上記でも全然問題ないのですが
設定ファイル(xcconfig)に集約していくのは如何でしょうか。

提案の理由としては`#if DEBUG` 節を多用しなくていいのでコードが読みやすくなるのと、
設定周りが一元管理されるため可読性が向上するためです。

一応参考までに手順記載しときました

## plistの修正
plist に `applicationKeyString`,  `clientKeyString`, `snsPlatformArn` 用の key/valueを追加 

以下参考
![2018-03-08 11 15 26](https://user-images.githubusercontent.com/22883147/37129486-80b27c04-22c2-11e8-8dc4-cf83683e9e9c.png)

## 各xcconfig に plistに追加したキー (上記例では$TEST_KEY) の値を設定する
以下は `Supporting Files/Config.Debug.xcconfig` に設定。
![2018-03-08 11 23 20](https://user-images.githubusercontent.com/22883147/37129597-1eefc5f2-22c3-11e8-8ce1-98111c410203.png)

同様に`Config.Adhoc.xcconfig` ,`Config.Release.xcconfig`でも TEST_KEY の値を設定

## plistから値を取得

上の例でplistから値を取り出すとしたら以下とかでしょうか。 
(BundleのExtensionとか定義したらもうちょっと読みやすくなるかな思います)

`let value = (Bundle.main.object(forInfoDictionaryKey: "TestKey") as? String) ?? ""`

## 注意点
xcconfigで設定した値が反映されないときは clean してみてください

## 参考記事
https://blog.nagisa-inc.jp/archives/818

SatoshiN303 commented:
@ist-ikeno
ほかは大体👍だとおもいますので上記二点ご確認よろしくお願い致します。

ist-ikeno commented:
@satoshin-IST レビューいただきありがとうございます! 上記2点修正いたします。

ist-ikeno committed:
[fix]Created by 池野 湧太 on 2018/02/05. → Created by XLAuthorNameに変更。

レビューフィードバック修正。
Skelton.swift で生成したときに実装者の名前で書き換えるため。

ist-ikeno committed:
[fix] 環境変数で切り分けていた各キー値をxcconfigで切り分け

#if DEBUG 節を多用しなくていいのでコードが読みやすくなるのと、設定周りが一元管理されるため可読性が向上するためにxcconfigに集約。
キーを切り替える場合は、Config.Debug.xcconfig, Config.Adhoc.xcconfig, Config.Release.xcconfigで値を設定する。

ist-ikeno commented:
@satoshin-IST フィードバックをいただいた点を修正しました。ご確認お願いします。

SatoshiN303 commented:
![LGTM](http://www.lgtm.in/p/ux1XG6tvs)
